### ISSUE TITLE
could this be used to have semantic search based on an ontology?

### ISSUE THREAD

**Issue by gebrits** [eea/eea.elasticsearch.river.rdf#13]
I'm looking for a way to let a domain specific ontology guide what results are returned. e.g.: searching for "piano bar" would also return documents for 'bar' since "piano bar " ISA 'bar"

Am I looking in the correct place?

**Comment by mihaibivol** [comment]
This plugin does an indexing of the triples you want to index and allows you to perform faceted queries.

For any of the indexing options you see in the README, the plugin takes triples matching your query
```?s ?p ?o``` and creates a document with the ```_id``` being ```?s``` and with a series of facets corresponding to unique ```?p``` having as value ```?o```.

Let's say your query returns:

 ?s | ?p | ?o 
-----|-----|----
thing1 | name | athing
thing1 | hasValue | 1
thing1 | hasValue | 2
thing1 | otherProp | thing1 property
thing2 | name | anotherthing
thing2 | hasValue | 1
thing2 | otherProp | thing2 property
thing2 | customProp | a thing2 custom property

You will have the following documents in ES

```json

{
  "thing1" : {
    "name": "thing1",
    "hasValue": [1, 2],
    "otherProp": "thing1 property"
  }, 
  "thing2": {
    "name": "thing2",
    "hasValue": [1],
    "otherProp": "thing2 property",
    "customProp": "a thing2 custom property"
  }
}
```

You have control over which triples to index from your ontology (via CONSTRUCT / SELECT / sync queries -- see README)
After that, given your ElasticSearch index you will have control over which facets you want to search in your app given the documents above.

Given the two places you can guide the search: index / results there are tons of possibilities to actually guide it. It won't actually do semantic magic, you have to make your decisions on the triples you index and the results you wish to display :)

**Comment by demarant** [comment]
@gebrits:
tl;dr 
No. 
This plugin does not provide/bring any "logic inference/ reasoning" or any ontology inteligence to elastic search. As Mihai well detailed explained, the plugin is purely focused on indexing triples from a sparqlend point. Although you can use it to index your ontologies (rdfs and owl) from a sparql endpoint, the plugin does not provide a "reasoning engine". The "reasoning engine" is part of the semantic store like Virtuoso. Elastic search does not provide reasoning engine either, it provides indexing and speed.
